src/core/public/http/types.ts

```diff
@@ -217,30 +217,56 @@ export interface HttpFetchOptions extends HttpRequestInit {
   headers?: HttpHeadersInit;
 
   /**
-   * When `true` the return type of {@link HttpHandler} will be an {@link IHttpResponse} with detailed request and
+   * Whether or not the request should include the "system API" header to differentiate an end user request from Kibana
+   * internal request.
+   * Can be read on the server-side using KibanaRequest#isSystemApi. Defaults to `false`.
+   */
+  asSystemApi?: boolean;
+
+  /**
+   * When `true` the return type of {@link HttpHandler} will be an {@link HttpResponse} with detailed request and
    * response information. When `false`, the return type will just be the parsed response body. Defaults to `false`.
    */
   asResponse?: boolean;
 }
 
+/**
+ * Similar to {@link HttpFetchOptions} but with the URL path included.
+ * @public
+ */
+export interface HttpFetchOptionsWithPath extends HttpFetchOptions {
+  /*
+   * The path on the Kibana server to send the request to. Should not include the basePath.
+   */
+  path: string;
+}
+
 /**
  * A function for making an HTTP requests to Kibana's backend. See {@link HttpFetchOptions} for options and
- * {@link IHttpResponse} for the response.
+ * {@link HttpResponse} for the response.
  *
  * @param path the path on the Kibana server to send the request to. Should not include the basePath.
  * @param options {@link HttpFetchOptions}
- * @returns a Promise that resolves to a {@link IHttpResponse}
+ * @returns a Promise that resolves to a {@link HttpResponse}
  * @public
  */
 export interface HttpHandler {
+  /** @deprecated */
   <TResponseBody = any>(path: string, options: HttpFetchOptions & { asResponse: true }): Promise<
-    IHttpResponse<TResponseBody>
+    HttpResponse<TResponseBody>
+  >;
+  <TResponseBody = any>(options: HttpFetchOptionsWithPath & { asResponse: true }): Promise<
+    HttpResponse<TResponseBody>
   >;
```

Comment:
I think I would just add `path` to `HttpFetchOptions` and document that the `(path:string, options: HttpFetchOptions)` signature(s) will ignore `options.path`. This simplifies both the interface and the actual implementation in a few places.

Following this Iogic I would even go to change `Fetch.fetch` signature from 
```typescript
pathOrOptions: string | HttpFetchOptionsWithPath,
options: HttpFetchOptions = {}
``` 
to just 
```typescript
options: HttpFetchOptions // WhichNowHasPath
```
 and only let the shorthands use the dual signature

```typescript
private shorthand(method: string): HttpHandler {
    return (pathOrOptions: string | HttpFetchOptionsWithPath, options: HttpFetchOptions = {}) => {
[...]
```

Reply:
Only allowing shorthands would break plugins migrating from kfetch to core.http.fetch. We want this to be a seamless migration since there are so many usages, so I will keep both signatures for all methods for now.